src/overlays/actors/ovl_Dm_Gm/z_dm_gm.c

```diff
@@ -190,34 +220,42 @@ Actor* func_80C24838(PlayState* play) {
     return foundActor;
 }
 
-void func_80C248A8(DmGm* this, PlayState* play) {
+void DmGm_Initialize(DmGm* this, PlayState* play) {
     if ((this->an4ObjectSlot > OBJECT_SLOT_NONE) && SubS_IsObjectLoaded(this->an4ObjectSlot, play) &&
         (this->msmoObjectSlot > OBJECT_SLOT_NONE) && SubS_IsObjectLoaded(this->msmoObjectSlot, play)) {
         ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 14.0f);
-        SkelAnime_InitFlex(play, &this->skelAnime, &object_an1_Skel_012618, NULL, this->jointTable, this->morphTable,
-                           OBJECT_AN1_LIMB_MAX);
+        SkelAnime_InitFlex(play, &this->skelAnime, &gAnju1Skel, NULL, this->jointTable, this->morphTable,
+                           ANJU1_LIMB_MAX);
 
         this->animIndex = DMGM_ANIM_NONE;
-        DmGm_ChangeAnim(this, play, DMGM_ANIM_0);
+        DmGm_ChangeAnim(this, play, DMGM_ANIM_SIT_IN_DISBELIEVE);
         this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
         Actor_SetScale(&this->actor, 0.01f);
-        this->unk_2AE |= 1;
-        this->actor.draw = func_80C25000;
+        this->stateFlags |= 1;
+        this->actor.draw = DmGm_Draw;
 
         if ((play->sceneId == SCENE_YADOYA) && (play->curSpawn == 4)) {
-            this->unk_2B4 = func_80C24838(play);
-            DmGm_ChangeAnim(this, play, DMGM_ANIM_1);
+            this->lookAtActor = DmGm_FindAnjusMotherActor(play);
+            DmGm_ChangeAnim(this, play, DMGM_ANIM_SIT);
             this->actionFunc = DmGm_DoNothing;
         } else {
-            this->actionFunc = DmGm_HandleCutscene;
+            this->actionFunc = DmGm_HandleCouplesMaskCs;
         }
     }
 }
 
-void DmGm_HandleCutscene(DmGm* this, PlayState* play) {
+void DmGm_HandleCouplesMaskCs(DmGm* this, PlayState* play) {
     s32 csAnimIndex[] = {
-        DMGM_ANIM_0, DMGM_ANIM_0, DMGM_ANIM_12, DMGM_ANIM_2,  DMGM_ANIM_4,
-        DMGM_ANIM_6, DMGM_ANIM_8, DMGM_ANIM_10, DMGM_ANIM_11, DMGM_ANIM_3,
+        /* 0 */ 0, // no cue
```

Comment:
Where does `cueId = 0` get translated to no cue in code?

Reply:
I looked at the extracted scene files for this cutscene and there's no cue 0. I don't think there's anything in code that guarantees anything about any cue id

You can see this the cues for this cutscene by searching for `CS_ACTOR_ACTION_LIST(0x22D` in the `assets/scenes/Z2_YADOYA/Z2_YADOYA.c` file.

Would be interesting to check if any cutscene uses cue 0 for anything

Reply:
Other reason why I think this entry should be 0 instead of an actual enum value is because of the array of EnAn

https://github.com/AngheloAlf/mm/blob/9c09615203ba7f3624e200aa137c9afdfccc2878/src/overlays/actors/ovl_En_An/z_en_an.c#L3226-L3237

EnAn:
```c
    s32 animIds[] = {
        /* 0 */ 0, // no cue // gAnju1HoldHandsAnim
        /* 1 */ ENAN_ANIM_SITTING_IN_DISBELIEVE,
        /* 2 */ ENAN_ANIM_34, // object_an4_Anim_004A78
        /* 3 */ ENAN_ANIM_25, // object_an4_Anim_006CC0
        /* 4 */ ENAN_ANIM_26, // object_an4_Anim_007E3C
        /* 5 */ ENAN_ANIM_28, // object_an4_Anim_0013C8
        /* 6 */ ENAN_ANIM_30, // object_an4_Anim_00353C
        /* 7 */ ENAN_ANIM_32, // object_an4_Anim_0060B4
        /* 8 */ ENAN_ANIM_33, // object_an4_Anim_00041C
        /* 9 */ ENAN_ANIM_HOLDING_HANDS,
    };
```

DmGm:
```c
    s32 csAnimIndex[] = {
        /* 0 */ 0, // no cue // gAnju1SitInDisbelieveAnim
        /* 1 */ DMGM_ANIM_SIT_IN_DISBELIEVE,
        /* 2 */ DMGM_ANIM_12, // object_an4_Anim_004A78
        /* 3 */ DMGM_ANIM_2, // object_an4_Anim_006CC0
        /* 4 */ DMGM_ANIM_4, // object_an4_Anim_006CC0
        /* 5 */ DMGM_ANIM_6, // object_an4_Anim_0013C8
        /* 6 */ DMGM_ANIM_8, // object_an4_Anim_00353C
        /* 7 */ DMGM_ANIM_10, // object_an4_Anim_0060B4
        /* 8 */ DMGM_ANIM_11, // object_an4_Anim_00041C
        /* 9 */ DMGM_ANIM_HOLD_HANDS,
    };
```

(Added the symbols for the unnamed enum entries)

Both arrays refer basically to the same animations in every entry except for the cueId 0. To me it seems like a change produced by not actually wanting to refer to anim 0, but just to have a placeholder because that cueId doesn't exist for this cutscene

Reply:
I'm not entirely convinced, I think a cueId being unused doesn't mean it doesn't exist. It seems you put a cueId = 0 in the command and it would work just fine?

But you're right in that there's no cueId 0 used anywhere in the entire repo. Maybe I'm missing somewhere that blocks cueId 0. For that reason, it feels more like a repo-wide issue and not one limited to this PR, so we can figure this out later, so that this won't block this PR.

Reply:
From the looks of it, seems like setting cueId to 0 it would work but it would play an animation that wouldn't make sense given the context (why would Anju start this cutscene by being kneeled holding hands with Kafei?)

Reply:
I don't think values represent where in the cutscene it plays. So cueId 0 or 1 wouldn't be the first cueId in a cutscene? That's specified by the start and end frames in the command